apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -169,3 +232,78 @@ sign_and_push_to_mempool(Tx) ->
     ok = aec_tx_pool:push(SignedTx),
     lager:debug("pushed; peek() -> ~p",
                 [pp(aec_tx_pool:peek(10))]).
+
+get_block(Fun, Req) ->
+    {arity, 0} = erlang:fun_info(Fun,arity),
+    get_block(Fun, no_params, Req).
+
+get_block(Fun, Params, Req) ->
+    case get_block_from_chain(Fun, Params) of
+        {ok, Block} ->
+            TxObjects = read_optional_bool_param('tx_objects', Req, false),
+            %% swagger generated code expects the Resp to be proplist or map
+            %% and always runs jsx:encode/1 on it - even if it is already
+            %% encoded to a binary; that's why we use
+            %% aec_blocks:serialize_to_map/1 instead of
+            %% aec_blocks:serialize_for_network/1
+            {SerializeFun, DataSchema} =
+                case TxObjects of
+                    true ->
+                        {fun aec_blocks:serialize_client_readable/1, <<"BlockWithTxs">>};
+                    false ->
+                        {fun aec_blocks:serialize_to_map/1, <<"BlockWithTxsHashes">>}
+                end,
+            Resp0 = aehttp_dispatch_ext:cleanup_genesis(SerializeFun(Block)),
+            % we add swagger's definition name to the object so the
+            % result could be validated against the schema
+            Resp = maps:put(data_schema, DataSchema, Resp0),
+            lager:debug("Resp = ~p", [pp(Resp)]),
+            {200, [], Resp};
+        {_Code, _, _Reason} = Err ->
+            Err
+    end.
+
+get_block_txs_count(Fun) ->
+    {arity, 0} = erlang:fun_info(Fun,arity),
+    get_block_txs_count(Fun, no_params).
+
+get_block_txs_count(Fun, Params) ->
+    case get_block_from_chain(Fun, Params) of
+        {ok, Block} ->
+            {200, [], #{count => length(aec_blocks:txs(Block))}};
+        {_Code, _, _Reason} = Err ->
+            Err
+    end.
+
+get_block_from_chain(Fun, Params) ->
+    Res =
```

Comment:
Another way of doing this would have been to wrap the arity-0 fun as:
`fun(_) -> Fun() end`. Then, there would be no need to check arity also when invoking the fun.